Source/WebKit/NetworkProcess/cache/NetworkCache.cpp

```diff
@@ -233,6 +233,11 @@ static UseDecision makeUseDecision(NetworkProcess& networkProcess, PAL::SessionI
     if (cachePolicyAllowsExpired(request.cachePolicy()))
         return UseDecision::Use;
 
+    // We could have cached a redirect without a fragment and now may have
+    // a fragment in the URL.
+    if (request.url().hasFragmentIdentifier() && entry.redirectRequest())
+        return UseDecision::NoDueToRequestContainingFragments;
```

Comment:
I thought that we were removing the fragment identifier before going to the network process. Can you check this?

Reply:
Doesn't look like it.

```
(lldb) p request.url()
(const WTF::URL) { http://localhost:10000/#key=foo }
```

Reply:
Added a breakpoint right at this line and dumped out the url output.

Reply:
ok